### PR TITLE
Avoid Java Temp Files in task creation

### LINKED ISSUE #6173: Server Errors when sending temporary files (JDK regression?)
## Context
Task creation with initial data (Trying to create 250 tasks based on uploaded zips)

## Expected Behavior
task annotation bases should be created with the initial data

## Current Behavior
124 tasks successfully created, 119 failed
I see occurences of
```
backend: 2022-04-27 15:59:56,798 [error] com.scalableminds.webknossos.datastore.rpc.RPCRequest - Error sending WS request to http://localhost:9000/tracings/volume/2c174c09-6ed3-4264-aa8b-3ca043a0a30c/initialData (ID: 425): /tmp/playtemp6106843044668276672/data.zip9932978374925623459
backend: akka.stream.impl.io.FileSource$$anon$2.preStart(IOSources.scala:68)
backend:     akka.stream.impl.fusing.GraphInterpreter.init(GraphInterpreter.scala:306)
backend:     akka.stream.impl.fusing.GraphInterpreterShell.init(ActorGraphInterpreter.scala:594)
backend:     akka.stream.impl.fusing.ActorGraphInterpreter.tryInit(ActorGraphInterpreter.scala:702)
backend:     akka.stream.impl.fusing.ActorGraphInterpreter.preStart(ActorGraphInterpreter.scala:751)
backend:     akka.actor.Actor.aroundPreStart(Actor.scala:548)
backend:     akka.actor.Actor.aroundPreStart$(Actor.scala:548)
backend:     akka.stream.impl.fusing.ActorGraphInterpreter.aroundPreStart(ActorGraphInterpreter.scala:691)
backend:     akka.actor.ActorCell.create(ActorCell.scala:643)
backend:     akka.actor.ActorCell.invokeAll$1(ActorCell.scala:513)
backend:     akka.actor.ActorCell.systemInvoke(ActorCell.scala:535)
backend:     akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:295)
backend:     akka.dispatch.Mailbox.run(Mailbox.scala:230)
backend:     java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
backend:     java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
backend:     java.base/java.lang.Thread.run(Thread.java:832)
```

and

```
backend: 2022-04-27 16:00:02,435 [error] akka.actor.RepointableActorRef - Error during preStart in [akka.stream.impl.io.FileSource$$anon$2-fileSource]: /tmp/playtemp6106843044668276672/data.zip7533579796494177102
backend: java.nio.file.NoSuchFileException: /tmp/playtemp6106843044668276672/data.zip7533579796494177102
backend: 	at akka.stream.impl.io.FileSource$$anon$2.preStart(IOSources.scala:68)
backend: 	at akka.stream.impl.fusing.GraphInterpreter.init(GraphInterpreter.scala:306)
backend: 	at akka.stream.impl.fusing.GraphInterpreterShell.init(ActorGraphInterpreter.scala:594)
backend: 	at akka.stream.impl.fusing.ActorGraphInterpreter.tryInit(ActorGraphInterpreter.scala:702)
backend: 	at akka.stream.impl.fusing.ActorGraphInterpreter.preStart(ActorGraphInterpreter.scala:751)
backend: 	at akka.actor.Actor.aroundPreStart(Actor.scala:548)
backend: 	at akka.actor.Actor.aroundPreStart$(Actor.scala:548)
backend: 	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundPreStart(ActorGraphInterpreter.scala:691)
backend: 	at akka.actor.ActorCell.create(ActorCell.scala:643)
```

## Steps to Reproduce the bug
Create hybrid task type

Create tasks, based on the zip of zips linked in https://discuss.webknossos.org/t/task-creation-failed/1723/3
(Note that this requires a dataset with the name mentioned in the nml files)

As discussed in https://github.com/TheHive-Project/TheHive/issues/1089 and https://github.com/hmrc/binding-tariff-trader-frontend/pull/329 this sounds like a regression in openjdk. I could not test OpenJDK 15 an up. (could not install 15on my system; 16 and up as are not yet supported by Play 2.8.)

It may be worth considering not using temporary files but passing the content in-memory

### PR DESCRIPTION
**Pull request by fm3**
- During task creation, we no longer use java’s own temporary file code (see reasons in linked issue below), but instead use a custom temporary file service.
- The initial data requests are no longer sent in parallel to the tracingstore but instead sequentially, to avoid overloading the tracingstore (new function Fox.serialSequenceBox)

### URL of deployed dev instance (used for testing):
- https://taskcreationnotempfiles.webknossos.xyz

### Steps to test:
- Create volume or hybrid task type
- try to create (hundreds of) tasks with initial volume data
- should not fail

### Issues:
- fixes #6173

------
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Needs tracingstore update after deployment
- [x] Ready for review

### CODE REVIEW

File: app/models/task/TaskCreationService.scala [hunk 1]
```diff
@@ -406,8 +398,9 @@ class TaskCreationService @Inject()(taskTypeService: TaskTypeService,
             case _          => savedId
           }
       }
-      volumeTracingIds: List[Box[Option[String]]] <- Fox.sequence(
-        requestedTasks.map(requestedTask => saveVolumeTracingIfPresent(requestedTask, tracingStoreClient)))
+      volumeTracingIds: List[Box[Option[String]]] <- Fox.serialSequence(requestedTasks) { requestedTask =>
```
Comment (jstriebel) [inline app/models/task/TaskCreationService.scala:401]: Please add a comment, that this is to reduce stress on the tracingstore.

File: app/oxalis/files/TempFileService.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.util.Random
+
+class TempFileService @Inject()(cleanUpService: CleanUpService)(implicit ec: ExecutionContext) extends LazyLogging {
+
+  private val tmpDir: Path = Paths.get(System.getProperty("java.io.tmpdir")).resolve("webKnossosTempFiles")
+
+  private val activeTempFiles = scala.collection.mutable.Set[(Path, Long)]()
+
+  cleanUpService.register("Clean up expired temporary files", 1 hour)(cleanUpExpiredFiles())
```
Comment (jstriebel) [inline app/oxalis/files/TempFileService.scala:21]: Can we also enforce to clean all files (regardless of expiry) when the application is exiting? Assuming that `java.io.tmpdir` is cleaned up by the os on shutdown, files might still pile up when restarting the application, as non-expired files are still there.

**Review by jstriebel — COMMENTED** [review]
Props for digging into this, I think the solution is good. Maybe we should also add a comment about the bug in the code (e.g. in `TempFileService`)? See my two comments below, besides this LGTM :+1:

**Comment by fm3** [comment]
Thanks for your feedback! I implemented the suggestions and also noticed another bug, where slashes in the prefix would lead to wrong file paths (so I am removing those now)

**Review by jstriebel — APPROVED** [review]
:shipit:

**Pull request merged**